subprojects/functional/src/main/java/org/gradle/internal/Either.java

```diff
@@ -34,9 +37,54 @@
         return new Right<>(value);
     }
 
-    public abstract <U> U fold(Function<L, U> l, Function<R, U> r);
+    /**
+     * Is this a left?
+     */
+    public abstract boolean isLeft();
```

Comment:
💄 I'd probably leave these `is*` predicates out because:
1. functionality is already covered by the `Optional` getters
2. avoid bean property name confusion with the getters `getLeft` / `getRight`

Reply:
Done.